test/functional/shared.js

```diff
@@ -78,9 +78,20 @@ function makeCleanupFn(client) {
 function withClient(client, operation, errorHandler) {
   const cleanup = makeCleanupFn(client);
 
+  function operationHandler(client) {
```

Comment:
The alternative to this change would be wrapping those tests in promises; if that's preferred lmk.